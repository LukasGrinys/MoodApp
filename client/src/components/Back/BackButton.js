import React from 'react';
import styles from './BackButton.module.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';
import { useTheme } from '../../contexts/ThemeContext';
import classNames from 'classnames';

const BackButton = () => {
    const darkTheme = useTheme();

    return (
        <div className={styles.container}>
            <div 
                className={classNames(
                    styles.button,
                    darkTheme && styles.dark
                )} 
                onClick={ () => {window.history.back()}}
            >
                <FontAwesomeIcon icon={faArrowLeft}/>
                <div className={styles.text}>
                    Go back
                </div>
            </div>
        </div>
        
    );
};

export default BackButton;