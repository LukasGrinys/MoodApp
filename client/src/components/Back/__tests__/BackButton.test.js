import React from 'react';
import BackButton from '../BackButton';
import { shallow } from 'enzyme';

const wrapper = shallow(<BackButton/>);

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Has Arrow icon', () => {
    expect(wrapper.find('FontAwesome').prop('name')).toBe('arrow-left');
});

test('Calls window.history.back when clicked', () => {
    window.history.back = jest.fn();
    const button = wrapper.find('.button');
    button.simulate('click');
    expect(window.history.back).toHaveBeenCalledTimes(1);
});