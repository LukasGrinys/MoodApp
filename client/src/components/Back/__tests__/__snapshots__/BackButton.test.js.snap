// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<div
  className="container"
>
  <div
    className="button"
    onClick={[Function]}
  >
    <FontAwesome
      name="arrow-left"
    />
    <div
      className="text"
    >
      Go back
    </div>
  </div>
</div>
`;
